src/AutoMapper/XpressionMapper/Extensions/MapperExtensions.cs

```diff
@@ -176,7 +176,18 @@ public static List<ParameterExpression> GetDestinationParameterExpressions(this
         public static Dictionary<Type, Type> AddTypeMapping<TSource, TDest>(this Dictionary<Type, Type> typeMappings)
             => typeMappings == null
                 ? throw new ArgumentException(Resource.typeMappingsDictionaryIsNull)
-                : typeMappings.AddTypeMapping(typeof(TSource), typeof(TDest));
+                : typeMappings.AddTypeMapping(typeof(TSource), typeof(TDest));
+
+        private static void AddUnderlyimgArrayElement(this Dictionary<Type, Type> typeMappings, Type sourceType, Type destType)
+        {
+            if (sourceType.IsArray && destType.IsArray)

```

Comment:
Maybe consider all collections and use [ElementTypeHelper](https://github.com/AutoMapper/AutoMapper/blob/524f587f51d43d3cb17aacfaeffb41b70f93c39c/src/AutoMapper/Mappers/Internal/ElementTypeHelper.cs)?